docs/other/guides/how-to-create-scaffolding-templates.md

```diff
@@ -0,0 +1,45 @@
+---
+id: creatingScaffoldingTemplates
+title: How to create Scaffolding Templates
+sidebar_label: Creating Scaffolding Templates
+---
+
+### Introduction
+As discussed in [Scaffolding](/docs/get-started/scaffolding), it is possible to save yourself a lot of time by using one of the default Scaffolding Templates.
+
+The great news are that you may also write your own templates! In this article we will go in depth about it.
+
+### Writing your first Scaffolding Template
+We use `node modules` for all of our templates. You should create a new package whose name follows the following pattern: `webiny-cli-scaffold-*`.
+
+For example, valid names could be:
+* webiny-cli-scaffold-append-unicorns
+* webiny-cli-scaffold-add-delicious-donuts-to-my-site
+
+Next, your package must export an array of [Plugins](/docs/deep-dive/plugins-crash-course) (a single Plugin is just as fine). Your Plugins must follow the object structure below:
+```
+{
+    name: String,           // Plugin info
+    description: String,    // Plugin info
+    scaffold: {
+        name: String        // Used in the Terminal
+        questions({ context }): Function:[InquirerQuestion]    // Can be just an array aswell
+        generate({ input, context, oraSpinner }): async Function  
+    }
+}
```

Comment:
This whole kinda-TS-type section: I would prefer if it was actual TS, with valid TS types. Do we have types for these template plugins? Would be great to actually have types in the `cli` package (I know that we don't use TS in that package, but types for plugins would be very very useful). Maybe add a `cli/types.ts` file, like all other projects do in `src/types.ts` (here we don't have `src` so  add it to the root of the package).

For `InquirerQuestion` you can install `@types/inquirer` package, that will provide you with inquirer TS declarations.

Reply:
done